src/server/helpers/file.py

```diff
@@ -45,3 +46,15 @@ def write_file(
     path = fs.get_path(fullpath)
 
     return path
+
+
+def is_gitignore(project: Project) -> Callable[[str], bool]:
```

Comment:
Nice solution with function factory! I would call it something like `file_filter = create_file_filter(project)` to make it explicit that you create a function